pkg/tsdb/legacydata/service/service.go

```diff
@@ -76,10 +124,15 @@ func (h *Service) HandleRequest(ctx context.Context, ds *models.DataSource, quer
 		Headers: query.Headers,
 	}
 
+	// Apply Configured Custom Headers to query request.
+	for k, v := range customHeaders(ds.JsonData, instanceSettings.DecryptedSecureJSONData) {
+		req.Headers[k] = v
```

Comment:
I added a check to make sure that custom headers cannot change headers that we have set (i.e. `From-Alert` and `X-Cache-Skip` in ngalert).

Reply:
There could be some headers we don't want them to change like the Host header, or for Hosted Metrics some kind of tracing headers? I'm not 100% sure.